src/os/bluestore/BlueStore.cc

```diff
@@ -6407,8 +6405,6 @@ int BlueStore::stat(
   bool allow_eio)
 {
   Collection *c = static_cast<Collection *>(c_.get());
-  if (!c->exists)
-    return -ENOENT;
```

Comment:
All of these are there because we expose a collection handle to the ObjectStore interface user, and the could go remove_collection on it and still have a handle.  I'm not sure that ever happens (it really shouldn't) but I don't think it's a good idea to remove these.